Temporarily revert https://github.com/rust-lang/libc/commit/5a3b5da3459b
and apply https://github.com/rust-lang/libc/commit/7175b7cce468

error[E0308]: mismatched types
   --> cargo-crates/nix-0.24.3/src/sys/event.rs:227:19
    |
227 |             data: data as type_of_data,
    |                   ^^^^^^^^^^^^^^^^^^^^ expected `i64`, found `isize`

error[E0063]: missing field `ext` in initializer of `kevent`
   --> cargo-crates/nix-0.24.3/src/sys/event.rs:222:26
    |
222 |         KEvent { kevent: libc::kevent {
    |                          ^^^^^^^^^^^^ missing `ext`

error[E0425]: cannot find value `NOTE_PCTRLMASK` in crate `libc`
    --> cargo-crates/nix-0.24.3/src/sys/event.rs:175:9
     |
 175 |         NOTE_PCTRLMASK;
     |         ^^^^^^^^^^^^^^
     |
    ::: cargo-crates/libc-0.2.190/src/unix/bsd/freebsdlike/freebsd/mod.rs:2264:1
     |
2264 | pub const NOTE_FFCTRLMASK: u32 = 0xc0000000;
     | ------------------------------ similarly named constant `NOTE_FFCTRLMASK` defined here
     |
    ::: cargo-crates/nix-0.24.3/src/macros.rs:70:27
     |
  70 |                     const $Flag = libc::$Flag $(as $cast)*;
     |                           ----- due to this macro variable
     |
help: a constant with a similar name exists
     |
 175 -         NOTE_PCTRLMASK;
 175 +         NOTE_FFCTRLMASK;
     |

--- cargo-crates/libc-0.2.190/build.rs.orig	2006-07-24 01:21:28 UTC
+++ cargo-crates/libc-0.2.190/build.rs
@@ -185,8 +185,8 @@ fn main() {
             // The ABI of libc used by std is backward compatible with FreeBSD 12.
             12
         } else {
-            // The ABI of libc from crates.io is backward compatible with FreeBSD 12.
-            12
+            // The ABI of libc from crates.io is backward compatible with FreeBSD 11.
+            11
         };
         // On CI, we detect the actual FreeBSD version and match its ABI exactly,
         // running tests to ensure that the ABI is correct.
--- cargo-crates/libc-0.2.190/src/unix/bsd/freebsdlike/freebsd/mod.rs.orig	2006-07-24 01:21:28 UTC
+++ cargo-crates/libc-0.2.190/src/unix/bsd/freebsdlike/freebsd/mod.rs
@@ -2280,6 +2280,9 @@ pub const NOTE_PDATAMASK: u32 = 0x000fffff;
 pub const NOTE_FORK: u32 = 0x40000000;
 pub const NOTE_EXEC: u32 = 0x20000000;
 pub const NOTE_PDATAMASK: u32 = 0x000fffff;
+#[deprecated(since = "1.0.0", note = "Not stable across OS versions")]
+#[doc(hidden)]
+pub const NOTE_PCTRLMASK: u32 = 0xf0000000;
 pub const NOTE_TRACK: u32 = 0x00000001;
 pub const NOTE_TRACKERR: u32 = 0x00000002;
 pub const NOTE_CHILD: u32 = 0x00000004;
